protocol/src/main/protobuf/Types.proto

```diff
@@ -73,6 +74,7 @@ message MajorType {
   optional int32 precision = 4; // used for decimal types
   optional int32 scale = 5; // used for decimal types
   optional int32 timeZone = 6; // used by TimeStamp type
+  repeated MinorType sub_type = 7; // used by Union type
```

Comment:
Why do we need this for Union when we don't need it for other complex types (LIST, MAP, ...)